src/Razor/test/Microsoft.VisualStudio.LanguageServerClient.Razor.Test/HtmlCSharp/SignatureHelpHandlerTest.cs

```diff
@@ -132,10 +144,46 @@ public async Task HandleRequestAsync_HtmlProjection_InvokesHtmlLanguageServer_Re
         public async Task HandleRequestAsync_CSharpProjection_InvokesCSharpLanguageServer_ReturnsItem()
         {
             // Arrange
+            var text = """
+                @code
+                {
+                    void M(int a, int b, int c)
+                    {
+                        M(
+                    }
+                }
+                """;
+            var cursorPosition = new Position { Line = 4, Character = 10 };
+
+            var documentUri = new Uri("C:/path/to/file.razor");
+            var csharpDocumentUri = new Uri("C:/path/to/file.razor__virtual.cs");
+            var codeDocument = CreateCodeDocument(text, documentUri.AbsoluteUri);
+
+            var csharpDocumentSnapshot = CreateCSharpVirtualDocumentSnapshot(codeDocument, csharpDocumentUri.AbsoluteUri);
+            var documentSnapshot = new TestLSPDocumentSnapshot(
+                documentUri,
+                version: 1,
+                snapshotContent: codeDocument.GetSourceText().ToString(),
+                csharpDocumentSnapshot);
+
+            var projectionResult = await TestLSPProjectionProvider.Instance.GetProjectionAsync(
+                documentSnapshot, cursorPosition, CancellationToken.None).ConfigureAwait(false);
+
+            var textDocumentIdentifier = new TextDocumentIdentifier { Uri = csharpDocumentUri };
+            var signatureHelpContext = new SignatureHelpContext { IsRetrigger = false, TriggerCharacter = "(", TriggerKind = SignatureHelpTriggerKind.TriggerCharacter };
+            var signatureHelpParams = new SignatureHelpParams { TextDocument = textDocumentIdentifier, Position = projectionResult.Position, Context = signatureHelpContext };
+
+            var result = await CSharpTestLspServerHelpers.ExecuteCSharpRequestAsync<SignatureHelpParams, SignatureHelp>(
```

Comment:
What if you were to create a TestLSPRequetsInvoker to utilize this? That way this test could potentially just call directly into the `SIgnatureHelpHandler` and you wouldn't need to call into the projection provider etc.?

Reply:
Good idea! Just updated